src/main/scala/com/fulcrumgenomics/bam/TrimPrimers.scala

```diff
@@ -178,12 +204,12 @@ class TrimPrimers
   }
 
   /** Trims all the reads for a given template. */
-  def trimReadsForTemplate(detector: AmpliconDetector, reads: Seq[SamRecord]): Unit = {
+  def trimReadsForTemplate(detector: AmpliconDetector, reads: Seq[SamRecord]): Boolean = {
     val rec1 = reads.find(r => r.paired && r.firstOfPair  && !r.secondary && !r.supplementary)
     val rec2 = reads.find(r => r.paired && r.secondOfPair && !r.secondary && !r.supplementary)
 
     val readsToClip = if (firstOfPair) reads.filter(_.firstOfPair) else reads
-
+    var skip = false
```

Comment:
Avoid using `var`s in scala as much as possible, unless it's very tightly scoped and needed for performance (e.g. faster for loops). I'll update with my PR